jetty-client/src/main/java/org/eclipse/jetty/client/HttpContent.java

```diff
@@ -81,6 +81,18 @@ public HttpContent(ContentProvider provider)
         this.iterator = provider == null ? Collections.<ByteBuffer>emptyIterator() : provider.iterator();
     }
 
+    /**
+     * 
+     * @param buffer
+     * @return whether the input buffer has been closed
+     */
+    private static boolean isBufferClosed(ByteBuffer buffer)
```

Comment:
The name and the comment are the wrong sense.  It is not if the buffer has been closed, but if the buffer *is the* `CLOSE` buffer, which is a sentinel (aka poison pill) value.

I'd prefer:
```
  * @return true if the buffer is the sentinel instance {@link CLOSE}
  */
private static boolean isTheCloseBuffer(ByteBuffer buffer)
```